airflow/models/baseoperator.py

```diff
@@ -826,6 +827,8 @@ def __init__(
         self.task_id = task_group.child_id(task_id) if task_group else task_id
         if not self.__from_mapped and task_group:
             task_group.add(self)
+        if isinstance(task_group, MappedTaskGroup):
+            self.deps = self.deps | {MappedTaskUpstreamDep()}
```

Comment:
Instead of doing this, I would choose to make `MappedTaskUpstreamDep` handle non-mapped task groups with no-op.

Reply:
I have now added `MappedTaskUpstreamDep` to the default deps of the `BaseOperator`. Please let me know what you think.